### ISSUE TITLE
Model Weights

### ISSUE THREAD

**Issue by Ashu7788-alt** [HamzaGbada/direct-neighbor-vrd#2]
您好，请问代码的模型中的 model_path=args.dataset + "_word_classification.pth",是要自己进行预训练吗？我运行代码的时候出现couldn't find it in the cached files and it looks like FUNSD_word_classification.pth is not the path to a directory containing a file named config.json.

**Comment by Ashu7788-alt** [comment]
你好，我还有个问题，我在训练的时候Train F1-score， Val F1-score， Train Accuracy等一直都是固定的值，并没有随着训练进程的进行而发生改变，请问这个是什么原因呢？

**Comment by HamzaGbada** [comment]
@Ashu7788-alt Can you please write the issue details in English?

**Comment by Ashu7788-alt** [comment]
> [@Ashu7788-alt](https://github.com/Ashu7788-alt) Can you please write the issue details in English?

Sorry, it was my oversight. I’m currently trying to run the code and had a couple of questions：

1、Model Weights Loading Issue
In the code, the model is loaded using the following line:
model_path = args.dataset + "_word_classification.pth"
Does this mean I need to pre-train the model and generate this .pth file myself?
When I try to run the code, I receive the following error:
couldn't find it in the cached files and it looks like FUNSD_word_classification.pth is not the path to a directory containing a file named config.json
Just to confirm — is this file supposed to be downloaded separately, or generated through a training step?

2、Training Metrics Not Changing
While training, I noticed that metrics such as Train F1-score, Validation F1-score, and Train Accuracy remain constant throughout the training process. They do not update or change at all.
Could you help me understand what might be causing this behavior? Is it possibly due to an issue in the dataset preprocessing, loss computation, or metrics calculation?

Any help or guidance would be greatly appreciated. Thank you in advance!

**Comment by HamzaGbada** [comment]
For the first question, we have trained using a custom word embedding model. You can check how to train it in the file [here](https://github.com/HamzaGbada/direct-neighbor-vrd/blob/master/train_word_embedding.py). If you're looking for a BERT example, you can explore that as well.

For the second question, the constant metrics during training can happen due to a variety of factors. The constant values depend on many parameters, and sometimes the GNN might face an over-smoothing problem, which could lead to this behavior. One thing that can help is using a good word embedding model, which will likely provide better results. Additionally, make sure that your GNN model isn't too complex, as a more complex model can sometimes exacerbate overfitting. To address the over-smoothing problem, you can refer to this paper: [over-smoothing GNN](https://arxiv.org/pdf/2303.10993).

Also, consider applying some regularization-like techniques as these can prevent over-smoothing and stabilize the training process.

**Comment by Ashu7788-alt** [comment]
Thank you very much for your response.

**Comment by Ashu7788-alt** [comment]
Hello, when I was reproducing your code, I didn't change the GNN model, but the evaluation metrics didn't update during reproduction. Why is that?

![Image](https://github.com/user-attachments/assets/89f5afa1-cad9-425e-b3fb-a59961bc1aa2)

When I make changes to the text embeddings, the evaluation metrics don't change either.

**Comment by HamzaGbada** [comment]
Did you know what does it mean Oversmoothing in GNN ?

**Comment by Ashu7788-alt** [comment]
> Did you know what does it mean Oversmoothing in GNN ?

Thank you for your response. I might have misunderstood your explanation earlier, but I had already identified the cause and resolved the issue yesterday afternoon.

**Comment by HamzaGbada** [comment]
> > Did you know what does it mean Oversmoothing in GNN ?
> 
> Thank you for your response. I might have misunderstood your explanation earlier, but I had already identified the cause and resolved the issue yesterday afternoon.

@Ashu7788-alt, Could you please provide you solution details and if possible put provide it as a pull request?
